pkg/application/registryConfigUseCase.go

```diff
@@ -0,0 +1,247 @@
+	var errs []error
+	var entriesToSet []*ecosystem.GlobalConfigEntry
+	var keysToDelete []common.GlobalConfigKey
+
+	for _, diff := range diffs {
+		switch diff.NeededAction {
+		case domain.ConfigActionSet:
+			entry := &ecosystem.GlobalConfigEntry{
+				Key:   diff.Key,
+				Value: common.GlobalConfigValue(diff.Expected.Value),
+			}
+			entriesToSet = append(entriesToSet, entry)
+		case domain.ConfigActionRemove:
+			keysToDelete = append(keysToDelete, diff.Key)
+		case domain.ConfigActionNone:
+			continue
+		default:
+			errs = append(errs, fmt.Errorf("cannot perform unknown action %q for global config with key %q", diff.NeededAction, diff.Key))
+		}
+	}
+
+	errs = append(errs, callIfNotEmpty(ctx, entriesToSet, useCase.globalConfigRepository.SaveAll))
+	errs = append(errs, callIfNotEmpty(ctx, keysToDelete, useCase.globalConfigRepository.DeleteAllByKeys))
+
+	return errors.Join(errs...)
+}
+
+func (useCase *EcosystemRegistryUseCase) applyDoguConfigDiffs(ctx context.Context, doguName common.SimpleDoguName, diffs domain.DoguConfigDiffs) error {
+	var errs []error
+	var entriesToSet []*ecosystem.DoguConfigEntry
+	var keysToDelete []common.DoguConfigKey
+
+	for _, diff := range diffs {
+		switch diff.NeededAction {
+		case domain.ConfigActionSet:
+			entry := &ecosystem.DoguConfigEntry{
+				Key:   common.DoguConfigKey{DoguName: doguName, Key: diff.Key.Key},
+				Value: common.DoguConfigValue(diff.Expected.Value),
+			}
+			entriesToSet = append(entriesToSet, entry)
+		case domain.ConfigActionRemove:
+			keysToDelete = append(keysToDelete, common.DoguConfigKey{DoguName: doguName, Key: diff.Key.Key})
+		case domain.ConfigActionNone:
+			continue
+		default:
+			errs = append(errs, doguUnknownConfigActionError(diff.NeededAction, diff.Key.Key, doguName))
+		}
+	}
+
+	errs = append(errs, callIfNotEmpty(ctx, entriesToSet, useCase.doguConfigRepository.SaveAll))
+	errs = append(errs, callIfNotEmpty(ctx, keysToDelete, useCase.doguConfigRepository.DeleteAllByKeys))
+
+	return errors.Join(errs...)
+}
+
+func (useCase *EcosystemRegistryUseCase) applySensitiveDoguConfigDiffs(ctx context.Context, doguName common.SimpleDoguName, diffs domain.SensitiveDoguConfigDiffs) error {
+	var errs []error
+
+	var entriesToSet []*ecosystem.SensitiveDoguConfigEntry
+	var keysToDelete []common.SensitiveDoguConfigKey
+
+	for _, diff := range diffs {
+		switch diff.NeededAction {
+		case domain.ConfigActionSet:
+			entry := &ecosystem.SensitiveDoguConfigEntry{
+				Key:   common.SensitiveDoguConfigKey{DoguConfigKey: common.DoguConfigKey{DoguName: doguName, Key: diff.Key.Key}},
+				Value: common.EncryptedDoguConfigValue(diff.Expected.Value),
+			}
+			// TODO: Check if dogu is installed
+			entriesToSet = append(entriesToSet, entry)
+		case domain.ConfigActionRemove:
+			keysToDelete = append(keysToDelete, common.SensitiveDoguConfigKey{DoguConfigKey: common.DoguConfigKey{DoguName: doguName, Key: diff.Key.Key}})
+		case domain.ConfigActionNone:
+			continue
```

Comment:
i introduced more specialized Actions cases:
`ConfigActionSetEncrypted` means that the config key needs to be encrypted.
`ConfigActionSetToEncrypt` means that the config key needs to be encrypted but another service needs to do this. This can happen if a dogu is not yet installed and therefore no encryption key pair is available.

The last one is fairly new in my branch for #43. Maybe you need to merge this into your branch first.